### PR TITLE
Rename pyaro related fields

### PR DESCRIPTION
**Pull request by magnusuMET**
## Change Summary

It might be confusing having `obs_config` nested in `obs_cfg`, so I suggest we rename this inner field to the more descriptive `pyaro_config`.

`data_id` is also not very descriptive, `reader_id` might give a better idea that it is related to the entrypoint in `pyaro`?

## Related issue number

Related to #1414 

## Checklist

* [x] Start with a draft-PR
* [x] The PR title is a good summary of the changes
* [x] PR is set to AeroTools and a tentative milestone
* [ ] Documentation reflects the changes where applicable
* [x] Tests for the changes exist where applicable
* [x] Tests pass locally
* [x] Tests pass on CI
* [x] At least 1 reviewer is selected
* [x] Make PR ready to review

### CODE REVIEW

File: pyaerocom/aeroval/setup_classes.py [hunk 1]
```diff
@@ -357,6 +358,8 @@ class EvalSetup(BaseModel):
 
     _aux_funs: dict = {}
 
+    obs_pyaro_readers: list[PyaroConfig] | None = None
```
Comment (lewisblake) [inline pyaerocom/aeroval/setup_classes.py]: Could this be `tuple[PyaroConfig, ...] | None = None`

**Review by thorbjoernl — COMMENTED** [review]
Have you checked if any of the base_configs need to be changed as a result of this change?

**Comment by magnusuMET** [comment]
> Have you checked if any of the base_configs need to be changed as a result of this change?

Grepping for `filename_or_obj_or_url`, which is a required field for `pyaro`, shows no base_configs have started using `pyaro`

**Review by thorbjoernl — APPROVED** [review]
:rocket:

**Pull request merged**